src/main/java/de/zalando/aruha/nakadi/domain/BatchItem.java

```diff
@@ -32,11 +35,20 @@ public void setStep(final EventPublishingStep step) {
         this.response.setStep(step);
     }
 
-    public void setPublishingStatus(final EventPublishingStatus publishingStatus) {
+    // avoid race condition with kafka producer callback
+    synchronized public void setPublishingStatus(final EventPublishingStatus publishingStatus) {
```

Comment:
I think there is no value in making this method `synchronized`. `setPublishingStatus(publishingStatus)` is already "almost" atomic. And if not it would be the responsibility of that method to synchronize.

What I meant in our discussion is to remove the setters for `publishingStatus` and `detail` and create one new method which is synchronized:

```
synchronized public void setPublishingStatusAndDetails(final EventPublishingStatus publishingStatus, final String cetail) {
...
}
```

That way it's avoided that the publishing status and the detail message are out of inconsistent.